<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.6.40">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="description" content="A collection of research papers, many of which were done primarily by undergraduate collaborators.">

<title>papers + projects ‚Äì Jo Hardin</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
</style>


<script src="../../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../../site_libs/clipboard/clipboard.min.js"></script>
<script src="../../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../../site_libs/quarto-search/fuse.min.js"></script>
<script src="../../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../../">
<link href="../../img/favicon.png" rel="icon" type="image/png">
<script src="../../site_libs/quarto-html/quarto.js"></script>
<script src="../../site_libs/quarto-html/popper.min.js"></script>
<script src="../../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../../site_libs/quarto-html/anchor.min.js"></script>
<link href="../../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../../site_libs/quarto-html/quarto-syntax-highlighting-9e1f2a4632985dffc2c80318c992baf5.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../../site_libs/bootstrap/bootstrap-3ecd1678aecfa8631a80e8a8e4c279e3.min.css" rel="stylesheet" append-hash="true" id="quarto-bootstrap" data-mode="light">
<script src="../../site_libs/quarto-contrib/iconify-1.0.0-beta.2/iconify-icon.min.js"></script>
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>


<link rel="stylesheet" href="../../styles.css">
</head>

<body class="nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg " data-bs-theme="dark">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container mx-auto">
    <a href="../../index.html" class="navbar-brand navbar-brand-logo">
    <img src="../../img/favicon.png" alt="" class="navbar-logo">
    </a>
    <a class="navbar-brand" href="../../index.html">
    <span class="navbar-title">Jo Hardin</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" role="menu" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll ms-auto">
  <li class="nav-item">
    <a class="nav-link" href="../../about.html"> 
<span class="menu-text">about</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../../courses.html"> 
<span class="menu-text">courses</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../../research.html"> 
<span class="menu-text">research</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../../projects.html"> 
<span class="menu-text">projects</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../../blogs.html"> 
<span class="menu-text">blogs</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../../letters.html"> 
<span class="menu-text">rec letters</span></a>
  </li>  
</ul>
          </div> <!-- /navcollapse -->
            <div class="quarto-navbar-tools">
</div>
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar zindex-bottom">
        
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">papers + projects</h1>
<p class="subtitle lead">so many fun projects, so little time</p>
</div>

<div>
  <div class="description">
    A collection of research papers, many of which were done primarily by undergraduate collaborators.
  </div>
</div>


<div class="quarto-title-meta">

    
  
    
  </div>
  


</header>


<figure class="figure">
<img src="featured.jpg" align="right" width="400" class="figure-img">
<figcaption>
From Lu and Hardin, 2021.
</figcaption>
</figure>
<p>
<a href="../../research/resume.pdf" target="_blank">Link to Jo Hardin‚Äôs CV</a>
</p>
<div style="text-align: start;">
<strong><strong>(links&nbsp;to preprints, arXiv, or journal website)</strong></strong>
</div>
<p>
<sup>#</sup> Indicates work done as an undergraduate student.
</p>
<h6 class="anchored">
<iconify-icon inline="" icon="fa-solid:chart-line"></iconify-icon> &nbsp; Statistical methods for high-throughput data
</h6>
<h6 class="anchored">
<iconify-icon inline="" icon="fa:sitemap"></iconify-icon> &nbsp; Inference in machine learning
</h6>
<h6 class="anchored">
<iconify-icon inline="" icon="fa-solid:chalkboard-teacher"></iconify-icon> &nbsp; Statistics and data science education
</h6>
<h6 class="anchored">
<iconify-icon inline="" icon="iconamoon:sign-equal-bold"></iconify-icon> &nbsp; Equity and flourishing in statistics and data science
</h6>
<p><br><br><br></p>
<!-- <li>  Hardin, J., <sup>#</sup>Quesada, L., <sup>#</sup>Ye, J., and Horton, N.J. <a href = "https://arxiv.org/abs/2406.07756" target = "_blank"><strong>The Exchangeability Assumption for Permutation Tests of Multiple Regression Models: Implications for Statistics and Data Science</strong></a>. In progress, 2026.</li>
<li>Hardin, J. and Horton, N.J. <strong>SQL in the Classroom</strong>. <em>Technology Innovations in Statistics Education</em> </li>
-->
<strong>2026</strong>
<ul>
<li>
<iconify-icon inline="" icon="fa-solid:chalkboard-teacher"></iconify-icon> Hardin, J. <a href="" target="_blank"><strong>Disciplinary Ethics in Data Science and Statistics</strong></a>, submitted, 2026.
</li>
<li>
<iconify-icon inline="" icon="fa-solid:chart-line"></iconify-icon> <sup>#</sup>Butler, S., <sup>#</sup>Choi, P., <sup>#</sup>Matheis, T., <sup>#</sup>Terdiman, M., and Hardin, J. <a href="https://github.com/hardin47/sicegar/blob/master/paper/paper.md" target="_blank"><strong>Modifications to sicegar: Analysis of Single-Cell viral Growth Curves</strong></a>, submitted, 2026.
</li>
<li>
<iconify-icon inline="" icon="fa:sitemap"></iconify-icon> <sup>#</sup>Kher, S. and Hardin, J. <a href="https://arxiv.org/abs/2505.18281" target="_blank"><strong>Measuring the Impact of Missingness in Traffic Stop Data</strong></a>. <em>Harvard Data Science Review</em>, accepted, 2026.
</li>
</ul>
<strong>2025</strong>
<ul>
<li>
<iconify-icon inline="" icon="fa-solid:chalkboard-teacher"></iconify-icon> Hardin, J. <a href="https://www.ams.org/journals/notices/202511/rnoti-p1276.pdf" target="_blank"><strong>A Mathematical Lens for Teaching Data Science</strong></a>. <em>Notices of the American Mathematical Society, 72</em>(11), 2025.
</li>
<li>
<iconify-icon inline="" icon="fa-solid:chart-line"></iconify-icon> <sup>#</sup>Colando, S., Schulz, D., and Hardin, J. <a href="https://doi.org/10.1093/bib/bbaf431" target="_blank"><strong>Selecting ChIP-Seq Normalization Methods from the Perspective of their Technical Conditions</strong></a>. <em>Briefings in Bioinformatics, 26</em>(4), 2025.
</li>
<li>
<iconify-icon inline="" icon="fa-solid:chart-line"></iconify-icon> Karnovsky, N.J., Warzybok, P., Cavalcanti, A., Hardin, J., <sup>#</sup>Brown, Z.W., <sup>#</sup>Caves, E. <sup>#</sup>Flynn, C.M., <sup>#</sup>Gallaher, G., <sup>#</sup>McDuffie, N., <sup>#</sup>McOmber, K., Bradley, R., Elliott, M., Saenz, B., and Jahncke, J. <a href="https://doi.org/10.3389/fbirs.2025.1587072" target="_blank"><strong>A Decade of Diving: Responses of Cassin‚Äôs auklets to variable foraging conditions in the California Current System</strong></a>. <em>Frontiers in Bird Science, 4</em>(1587072), 2025. .
</li>
</ul>
<strong>2024</strong>
<ul>
<li>
<iconify-icon inline="" icon="iconamoon:sign-equal-bold"></iconify-icon> Hardin, J. <a href="https://doi.org/10.1080/09332480.2024.2348978" target="_blank"><strong>CURV - connecting, uplifting, and recognizing voices</strong></a>, <em>Chance Magazine, Statistics Education Column, Taking a Chance in the Classroom 37</em>(2), 2024.
</li>
<li>
<iconify-icon inline="" icon="fa-solid:chalkboard-teacher"></iconify-icon> <sup>#</sup>Colando, S., Hardin, J. <a href="https://doi.org/10.1080/26939169.2024.2394542" target="_blank"><strong>Philosophy within Data Science Ethics Courses</strong></a>, <em>Journal of Statistics and Data Science Education 32</em>(4), 2024.
</li>
<li>
<sup>#</sup>Cruz, M., <sup>#</sup>Wei, A., Hardin, J., Radunskaya, A. <a href="https://doi.org/10.1080/10236198.2024.2316379" target="_blank"><strong>Long-term Averages of the Stochastic Logistic Map</strong></a>, <em>Journal of Difference Equations and Applications 30</em>(6), 2024.
</li>
<li>
<iconify-icon inline="" icon="fa-solid:chalkboard-teacher"></iconify-icon> √áetinkaya-Rundel, M., Hardin, J. <a href="https://www.openintro.org/book/ims/" target="_blank"><strong>Introduction to Modern Statistics</strong></a> OpenIntro, 2nd edition, 2024.
</li>
</ul>
<strong>2023</strong>
<ul>
<li>
<iconify-icon inline="" icon="fa-solid:chart-line"></iconify-icon> <sup>#</sup>Adams, J., <sup>#</sup>Hoang, J., <sup>#</sup>Petroni, E., <sup>#</sup>Ashby, E., Hardin, J., Stoebel, D. <a href="https://doi.org/10.1128/msystems.00663-23" target="_blank"><strong>The timing of transcription of RpoS-dependent genes varies across multiple stresses in <i>Escherichia coli</i> K-12</strong></a>, <em>mSystems 8</em>(5), 2023.
</li>
<li>
<iconify-icon inline="" icon="fa-solid:chart-line"></iconify-icon> <sup>#</sup>Ashby, E., <sup>#</sup>Havens, J., Hardin, J., Schulz, D. <a href="https://journals.asm.org/doi/10.1128/spectrum.00147-23" target="_blank"><strong>Chemical inhibition of bromodomain proteins in insect stage African trypanosomes perturbs silencing of the Variant Surface Glycoprotein repertoire and results in widespread changes in the transcriptome</strong></a>, <em>Microbiology Spectrum 11</em>(3), 2023.
</li>
<li>
<iconify-icon inline="" icon="iconamoon:sign-equal-bold"></iconify-icon> Hardin, J., Shahriari, S. <a href="https://doi.org/10.1080/10511970.2022.2082611" target="_blank"><strong>Community, Collaboration, and Climate</strong></a>, <em>PRIMUS: Problems, Resources, and Issues in Mathematics Undergraduate Studies, 33</em>(5), 2023.
</li>
</ul>
<strong>2022</strong>
<ul>
<li>
<iconify-icon inline="" icon="fa-solid:chalkboard-teacher"></iconify-icon> √áetinkaya-Rundel, M., Hardin, J., Baumer, B., McNamara, A., Horton, N., Rundel, C. <a href="https://escholarship.org/uc/item/7kk4d922" target="_blank"><strong>An Educator‚Äôs Perspective on the Tidyverse</strong></a>, <em>Technology Innovations in Statistics Education, 14</em> 2022.
</li>
<li>
<iconify-icon inline="" icon="fa-solid:chart-line"></iconify-icon> <sup>#</sup>Ashby, E., <sup>#</sup>Paddock, L., <sup>#</sup>Rollosson, L., <sup>#</sup>Tang, E., <sup>#</sup>Miller, G., <sup>#</sup>Wade, S., <sup>#</sup>Betts, H., <sup>#</sup>Porter, A., <sup>#</sup>Saada, C., Hardin, J., Schulz, D. <a href="https://doi.org/10.1128/msphere.00023-22" target="_blank"><strong>Genomic occupancy of the bromodomain protein Bdf3 is dynamic during differentiation of African trypanosomes from bloodstream to procyclic forms</strong></a>, <em>mSphere, 7</em> (3), 2022.
</li>
</ul>
<strong>2021</strong>
<ul>
<li>
<iconify-icon inline="" icon="fa-solid:chalkboard-teacher"></iconify-icon> √áetinkaya-Rundel, M., Hardin, J. <a href="https://www.openintro.org/book/ims/" target="_blank"><strong>Introduction to Modern Statistics</strong></a> OpenIntro, 1st edition, 2021.
</li>
<li>
<iconify-icon inline="" icon="fa-solid:chalkboard-teacher"></iconify-icon> Kim, A.Y., Hardin, J. <a href="https://www.tandfonline.com/doi/full/10.1080/10691898.2020.1799728" target="_blank"><strong>‚ÄúPlaying the whole game‚Äù: A data collection and analysis exercise with Google Calendar</strong></a>, <em>Journal of Statistics and Data Science Education, 29</em>(S1), 2021.
</li>
<li>
<iconify-icon inline="" icon="fa:sitemap"></iconify-icon> <sup>#</sup>Lu, B., Hardin, J. <a href="https://jmlr.org/papers/v22/18-558.html" target="_blank"><strong>A Unified Framework for Random Forest Prediction Error Estimation</strong></a>, <em>Journal of Machine Learning Research, 22</em>(8), 2021.
<ul>
<li>
<a href="http://computationalgenomics.bioinformatics.ucla.edu/portfolio-item/jo-hardin-prediction-intervals-for-random-forests-with-applications-to-high-throughput-data/" target="_blank"><b></b>Prediction intervals for random forests with applications to high throughput data&nbsp;</a><a href="https://youtu.be/x7F1Z8pGS6Q" target="_blank" rel="noopener noreferrer"><img src="https://i1.wp.com/computationalgenomics.bioinformatics.ucla.edu/wp-content/uploads/sites/11/2016/08/icon-youtube.gif?w=1200" alt=""></a>(Computational Genomics Summer Institute @ IPAM, 2017)
</li>
</ul>
</li>
<li>
<iconify-icon inline="" icon="fa-solid:chalkboard-teacher"></iconify-icon> Hardin, J. <strong>9 out of 10 Seniors Recommend this First-Year Seminar: Statistics in the World</strong>, In <a href="https://www.maa.org/press/ebooks/mathematical-themes-in-a-first-year-seminar" target="_blank"><em>Mathematical Themes in a First-Year Seminar</em></a>, eds.&nbsp;J. Schaefer, J. Bowen, M. Kozek, and P. Pierce,<span class="Apple-converted-space">&nbsp; </span>MAA Notes Series; 2021.
</li>
</ul>
<strong>2020</strong>
<ul>
<li>
<iconify-icon inline="" icon="iconamoon:sign-equal-bold"></iconify-icon> Hardin, J., Haushalter, K., Yong, D.<a href="https://new.seceij.net/articletype/projectreport/turning-stem-education-inside-out/" target="_blank"><strong>Turning STEM Education Inside-Out: Teaching and Learning Inside of Prisons</strong></a>, <em>Science Education and Civic Engagement: An International Journal, 12 </em>(2); 2020.
</li>
<li>
<sup>#</sup>Allison, K., <sup>#</sup>Hallman, M., <sup>#</sup>Koskelo E., Radunskaya, A., Hardin, J., Hudgings, J. <a href="https://aip.scitation.org/doi/10.1063/1.5135922"><strong>Increasing the speed of CCD-based thermoreflectance imaging</strong></a>, <em>Review of Scientific Instruments, 91</em>: 044901, 2020. https://doi.org/10.1063/1.5135922.
</li>
<li>
<iconify-icon inline="" icon="fa-solid:chalkboard-teacher"></iconify-icon> Baumer, B., Bray, A., √áetinkaya-Rundel, M., and Hardin, J.<span class="Apple-converted-space">&nbsp; </span><a href="https://www.tandfonline.com/doi/full/10.1080/10691898.2020.1730734" target="_blank"><strong>Teaching Introductory Statistics with DataCamp</strong></a>, <em>Journal of Statistics Education, 28(1)</em>; 89-97, 2020.
</li>
</ul>
<strong>2019</strong>
<ul>
<li>
<iconify-icon inline="" icon="fa-solid:chalkboard-teacher"></iconify-icon> Fiksel, J., Jager, L., Hardin, J., Taub, M. <strong><a href="https://tandfonline.com/doi/full/10.1080/10691898.2019.1617089" target="_blank">Using GitHub Classroom To Teach Statistics</a></strong>, <em>Journal of Statistics Education, 27</em>(2):&nbsp;<span class="page_range">110-119, 2019.</span>
</li>
<li>
<iconify-icon inline="" icon="fa:sitemap"></iconify-icon> Duron, C., Pan, Y., D. Gutmann, Hardin, J., Radunskaya, A.<a href="https://link.springer.com/article/10.1007/s11538-018-0526-z" target="_blank"><strong>Variability of Betweenness Centrality and Its Effect on Identifying Essential Genes</strong></a>, <em>Bulletin of Mathematical Biology, 81</em>(9): 3655-3673, 2019. (also <a href="https://link.springer.com/epdf/10.1007/s11538-018-0526-z?author_access_token=tkw6_ElsPYMBBKul8QM5Pve4RwlQNchNByi7wbcMAY46Et6dinaCO1JuBnTj0L7Q33M_vLO7sQS2GSZTuo_brh50lftaH7uBdRfV792EFC_OlKQ4A-hsCdV6kJ4vRQcTRqOPi9F8US_PKsLM-ivrwg%3D%3D" target="_blank">here</a>)
</li>
</ul>
<strong>2018</strong>
<ul>
<li>
<iconify-icon inline="" icon="fa-solid:chalkboard-teacher"></iconify-icon> Horton, N., Hardin, J. <strong><a href="http://iase-web.org/icots/10/proceedings/pdfs/ICOTS10_3A3.pdf" target="_blank">Challenges and Opportunities for Statistics and Data Science Undergraduate Major and Minor Degree Programs</a></strong>.&nbsp;<em>Proceedings of&nbsp;the Tenth&nbsp;International Conference on Teaching Statistics</em>, 2018.
</li>
<li>
<iconify-icon inline="" icon="fa-solid:chart-line"></iconify-icon> <sup>#</sup>Evans, C., Hardin, J., Stoebel, D.<strong><a href="https://academic.oup.com/bib/article/3056951/Selecting-between-sample-RNA-Seq-normalization" target="_blank"> Selecting between-sample RNA-Seq normalization methods from the perspective of their assumptions.</a></strong>&nbsp;<em>Briefings in Bioinformatics, 19</em>(5): 776‚Äì792, 2018. &nbsp;(<a href="https://arxiv.org/abs/1609.00959" target="_blank">paper @ arxiv.org</a>)
<ul>
<li>
<a href="http://computationalgenomics.bioinformatics.ucla.edu/portfolio-item/jo-hardin-assumptions-in-normalizing-rnaseq-data/" target="_blank">Tutorial on RNASeq Normalization and Differential Expression</a> (Computational Genomics Summer Institute @ IPAM, 2016)
</li>
<li>
<a href="http://computationalgenomics.bioinformatics.ucla.edu/portfolio-item/jo-hardin-tutorial-on-rnaseq-normalization-and-differential-expression/" target="_blank">Assumptions in Normalizing RNASeq Data</a> (Computational Genomics Summer Institute @ IPAM, 2016)
</li>
</ul>
</li>
<li>
<iconify-icon inline="" icon="iconamoon:sign-equal-bold"></iconify-icon> Hardin, J. <strong><a href="http://scholarship.claremont.edu/jhm/vol8/iss1/17/" target="_blank">Fun, Not Competition: The Story of My Math Club</a></strong>, <em>Journal of Humanistic Mathematics</em>, 8(1): 350-358, 2018.
</li>
<li>
<iconify-icon inline="" icon="fa-solid:chart-line"></iconify-icon> Pan, Y., Duron, C., Bush, E., Sims, P., Hardin, J., Radunskaya, A., Gutmann, D.&nbsp;<a href="http://journals.plos.org/plosone/article?id=10.1371/journal.pone.0190001" target="_blank"><strong>Graph Complexity Analysis Identifies an ETV5 Tumor-Specific Network in Human and Murine Low-Grade Glioma</strong></a>, <em>PLoS ONE</em>, 13(5): e0190001, 2018.&nbsp;
</li>
<li>
<iconify-icon inline="" icon="fa-solid:chalkboard-teacher"></iconify-icon> Hardin, J.&nbsp; <strong><a href="https://escholarship.org/uc/item/13g5g3dm" target="_blank">Dynamic Data in the Statistics Classroom</a></strong>.&nbsp;<em>Technological Innovations in Statistics Education, 11(1)</em>, 2018.&nbsp; (<a href="http://arxiv.org/abs/1603.04912" target="_blank">paper @ arxiv.org</a>, <a href="https://github.com/hardin47/DynamicData" target="_blank">full worked-out examples</a>)
<ul>
<li>
<a href="https://www.causeweb.org/cause/ecots/ecots16/posters/a/7" target="_blank">Dynamic Data in the Classroom</a> (eCOTS 2016)
</li>
<li>
<a href="https://channel9.msdn.com/events/useR-international-R-User-conference/useR2016/Dynamic-Data-in-the-Statistics-Classroom" target="_blank">Dynamic Data in the Statistics Classroom</a> (useR 2016)
</li>
</ul>
</li>
</ul>
<strong>2017</strong>
<ul>
<li>
<iconify-icon inline="" icon="fa-solid:chart-line"></iconify-icon> <sup>#</sup>Wong, G., Bonocora, R., <sup>#</sup>Schep, A., <sup>#</sup>Beeler, S., Lee, A., <sup>#</sup>Shull, L., <sup>#</sup>Batachari, L., <sup>#</sup>Dillon, M., <sup>#</sup>Evans, C., <sup>#</sup>Becker, C., Bush, E., Hardin, J., Wade, J., Stoebel, D. <strong><a href="http://jb.asm.org/content/199/7/e00755-16.abstract" target="_blank">The genome-wide transcriptional response to varying RpoS levels in <em>Escherichia coli</em> K-12.</a></strong> <em>Journal of Bacteriology</em>, 199:e00755-16, 2017. &nbsp; (<a href="http://biorxiv.org/content/early/2016/10/21/082537" target="_blank">paper @ biorxiv.org</a>)
</li>
<li>
<iconify-icon inline="" icon="fa-solid:chalkboard-teacher"></iconify-icon> Hardin, J., Kloke, J. ‚ÄúStatistical Analyses‚Äù in <strong>Current Protocols in Molecular Biology</strong>, Appendix 4A, John Wiley &amp; Sons, 2017.
</li>
<li>
<iconify-icon inline="" icon="fa-solid:chart-line"></iconify-icon> <sup>#</sup>Evans, C., Hardin, J., Huber, M., Stoebel, D., <sup>#</sup>Wong, G. <strong> <a href="http://arxiv.org/abs/1410.3370" target="_blank">Differential expression analysis for multiple conditions</a></strong>,&nbsp;unpublished, 2017. (<a href="http://arxiv.org/abs/1410.3370" target="_blank">paper @ arxiv.org</a>)
</li>
</ul>
<strong>2016</strong>
<ul>
<li>
<iconify-icon inline="" icon="fa:sitemap"></iconify-icon> <sup>#</sup>Coleman, J., <sup>#</sup>Replogle, J., Chandler, G., Hardin, J. <strong><a href="https://www.degruyter.com/view/j/sagmb.2016.15.issue-2/sagmb-2014-0081/sagmb-2014-0081.xml" target="_blank">Resistant Multiple Sparse Canonical Correlation.</a></strong> <em>Statistical Applications in Genetics and Molecular Biology;15</em> (2): 123-38, 2016. (<a href="http://arxiv.org/abs/1410.3355" target="_blank">paper @ arxiv.org</a>)
</li>
</ul>
<strong>2015</strong>
<ul>
<li>
<iconify-icon inline="" icon="fa-solid:chalkboard-teacher"></iconify-icon> Hardin, J., Hoerl, R., Horton, N.J., Nolan, D. <strong><a href="http://www.tandfonline.com/doi/full/10.1080/00031305.2015.1077729" target="_blank">Data Science in Statistics Curricula: Preparing Students to ‚ÄòThink with Data‚Äô.</a></strong> <em>The American Statistician, 69(4):343-353, 2015</em>. (<a href="http://arxiv.org/abs/1410.3127" target="_blank">paper @ arxiv.org</a>)
</li>
<li>
<iconify-icon inline="" icon="fa:sitemap"></iconify-icon> Hardin, J., Sarkis, G., <sup>#</sup>URC, P.C. <strong><a href="http://www.amstat.org/publications/jse/v23n2/hardin.pdf" target="_blank">Network Analysis with the Enron Email Corpus.</a></strong> <em>Journal of Statistics Education, 23(2)</em>, 2015. (P.C. URC stands for the Pomona College Undergraduate Research Circle whose members for this project were Timothy Kaye, David Khatami, Daniel Metz, and Emily Proulx.) (<a href="http://arxiv.org/abs/1410.2759" target="_blank">paper @ arxiv.org</a>)
</li>
</ul>
<strong>2013</strong>
<ul>
<li>
<iconify-icon inline="" icon="fa:sitemap"></iconify-icon> Hardin, J., Garcia, S.R., Golan, D. <strong><a href="https://projecteuclid.org/journals/annals-of-applied-statistics/volume-7/issue-3/A-method-for-generating-realistic-correlation-matrices/10.1214/13-AOAS638.full" target="_blank"> A method for generating realistic correlation matrices</a> </strong>, <em>Annals of Applied Statistics, 7</em>: 1733-1762, 2013.
</li>
</ul>
<strong>2012</strong>
<ul>
<li>
<iconify-icon inline="" icon="fa-solid:chalkboard-teacher"></iconify-icon> <sup>#</sup>Brieger, K, J. Hardin. <strong><a href="https://hardin47.netlify.app/research/papers/files/Brieger3.pdf" target="_blank"> Medicine and Statistics: the inextricable link </a> </strong> <em> Chance, 25</em>: 31-34, 2012.
</li>
<li>
<sup>#</sup>Head, A., Hardin, J., Adolph, S. <strong><a href="http://pages.pomona.edu/~jsh04747/Research/Head%20et%20al%20EES%20v3.pdf" target="_blank">New methods for estimating maximum performance and the correlation of sample measures</a></strong>, <em>Environmental and Ecological Statistics, 19</em>: 127-137, 2012.
</li>
</ul>
<strong>2011</strong>
<ul>
<li>
Karnovsky, N.J., <sup>#</sup>Brown, Z.W.,&nbsp;Welcker, J., Harding, A.M.A.,&nbsp;Walkusz, W.,&nbsp;Cavalcanti, A.,&nbsp;Hardin, J.,&nbsp;Kitaysky, A.,&nbsp;Gabrielsen, G.,&nbsp;Gr√©millet, D.&nbsp;<strong><a href="http://www.int-res.com/articles/meps_oa/m440p229.pdf" target="_blank">Inter-colony comparison of diving behavior of an Arctic top predator: implications for warming in the Greenland Sea</a></strong>,&nbsp;<em>Marine Ecology Progress Series, 440:</em>&nbsp;229-240, 2011.&nbsp;&nbsp;DOI: 10.3354/meps0935.
</li>
<li>
Grosfils, E.B., <sup>#</sup>Long, S.M., <sup>#</sup>Venechuk, E.M.,<sup>#</sup> Hurwitz, D.M., <sup>#</sup>Richards, J.W., <sup>#</sup>Kastl, Brian, <sup>#</sup>Drury, D.E., Hardin, J., 2011,&nbsp;<a href="http://astrogeology.usgs.gov/Projects/PlanetaryMapping/MapStatus/VenusStatus/V14.html" target="_blank"><strong>Geologic map of the&nbsp;Ganiki&nbsp;Planitia&nbsp;quadrangle (V-14), Venus: U.S. Geological Survey Scientific Investigations Map 3121</strong></a>.Now available at USGS as an <a href="https://usgs.maps.arcgis.com/apps/webappviewer/index.html?id=c14edac7e91b4fc5ade4042e3036ec58" target="_blank">interactive map</a>.
</li>
</ul>
<strong>2010</strong>
<ul>
<li>
<iconify-icon inline="" icon="fa:sitemap"></iconify-icon> <sup>#</sup>Richards, J., Hardin, J.,&nbsp;Grosfils, E.&nbsp;<strong><a href="http://pages.pomona.edu/~jsh04747/Research/WeightedMBClustering_CompGeo_final.pdf" target="_blank">Weighted Model-Based Clustering for Remote Sensing Image Analysis</a></strong>,<em>&nbsp;Computational Geosciences, 14</em>: 125-136, 2010.
</li>
</ul>
<p>&nbsp;</p>
<strong>2009</strong>
<ul>
<li>
<iconify-icon inline="" icon="fa-solid:chart-line"></iconify-icon> Hardin, J.,&nbsp;Wilson, J.&nbsp;<strong><a href="http://biostatistics.oxfordjournals.org/cgi/content/full/10/3/446" target="_blank">A note on&nbsp;oligonucleotide&nbsp;expression values not being normally distributed</a></strong>,&nbsp;<em>Biostatistics</em>,&nbsp;<em>10</em>: 446-450, 2009. (full&nbsp;manuscript at&nbsp;<a href="http://pages.pomona.edu/~jsh04747/Research/fullpaper_supplement.pdf" target="_blank">Supplementary Material to A note onoligonucleotide&nbsp;expression values not being normally distributed</a>)
</li>
</ul>
<p>&nbsp;</p>
<strong>2008</strong>
<ul>
<li>
<iconify-icon inline="" icon="fa-solid:chart-line"></iconify-icon> <sup>#</sup>Yiu, G., <sup>#</sup>McCord, A., <sup>#</sup>Wise, A.,&nbsp;&nbsp;<sup>#</sup>Jindal, R., <sup>#</sup>Hardee, J.,&nbsp;<sup>#</sup>Kuo, A., <sup>#</sup>Yuen&nbsp;Shimogawa, M.,&nbsp;Cahoon, L., Wu, M., Kloke, J., Hardin, J., Mays&nbsp;Hoopes, L.L.;&nbsp;<strong><a href="http://pages.pomona.edu/~jsh04747/Research/yeastSep07.pdf" target="_blank">Pathways Change in Expression During&nbsp;Replicative&nbsp;Aging in&nbsp;<em>Saccharomyces</em><em>&nbsp;cerevisiae</em></a></strong>,&nbsp;<em>Journal of Gerontology, 63A</em>: 21-34, 2008.
</li>
</ul>
<p>&nbsp;</p>
<strong>2007</strong>
<ul>
<li>
<iconify-icon inline="" icon="fa-solid:chart-line"></iconify-icon> Hardin, J.,&nbsp;<sup>#</sup>Mitani, A., <sup>#</sup>Hicks, L.,&nbsp;<sup>#</sup>VanKoten, B.;&nbsp;<strong><a href="http://www.biomedcentral.com/content/pdf/1471-2105-8-220.pdf" target="_blank">A Robust Measure of Correlation&nbsp;Between&nbsp;Two Genes on a Microarray</a></strong>,&nbsp;<em>BMC Bioinformatics</em>,&nbsp;<em>8</em>:220, 2007. (R code:&nbsp;<a href="http://pages.pomona.edu/~jsh04747/Research/biwt.r" target="_blank">biwt.r</a>)
</li>
<li>
Adolph, S., Hardin, J.;&nbsp;<strong><a href="http://pages.pomona.edu/~jsh04747/Research/Adolph&amp;Hardin_official.pdf" target="_blank">Estimating Phenotypic Correlations: Correcting for Bias Due to&nbsp;Intraindividual&nbsp;Variability</a></strong>,&nbsp;<em>Functional Ecology</em>,&nbsp;<em>21</em>: 178-184, 2007.
</li>
</ul>
<p>&nbsp;</p>
<strong>2006</strong>
<ul>
<li>
<iconify-icon inline="" icon="fa-solid:chart-line"></iconify-icon> <sup>#</sup>Wise, A., Hardin, J.,&nbsp;Hoopes, L.;&nbsp;<strong><a href="http://pages.pomona.edu/~jsh04747/Research/chancepaper3.pdf" target="_blank">Yeast&nbsp;Through&nbsp;the Ages: a statistical analysis of genetic changes in aging yeast</a></strong>,&nbsp;<em>Chance</em>,&nbsp;<em>19</em>, 39-44, 2006.
</li>
<li>
<iconify-icon inline="" icon="fa-solid:chart-line"></iconify-icon> Hardin, J.,&nbsp;Hoopes, L., <sup>#</sup>Murphy, R.;&nbsp;<strong><a href="http://www.stat.auckland.ac.nz/~iase/publications/17/3E1_HARD.pdf" target="_blank">Analyzing DNA Microarrays with Undergraduate Statisticians</a></strong>,&nbsp;<em>Proceedings of&nbsp;the Seventh&nbsp;International Conference on Teaching Statistics</em>, 2006.
</li>
</ul>
<p>&nbsp;</p>
<strong>2005</strong>
<ul>
<li>
<iconify-icon inline="" icon="fa-solid:chalkboard-teacher"></iconify-icon> Altman, N., Banks, D., Hardwick, J., Roeder, K.,&nbsp;Craigmile, P., Hardin, J., Gupta, M.&nbsp;&nbsp;<a href="http://pages.pomona.edu/~jsh04747/Research/NRSG%20web.pdf" target="_blank"><strong>The IMS New Researchers‚Äô Survival Guide</strong></a>, Institute of Mathematical Statistics; 2005.
</li>
<li>
<iconify-icon inline="" icon="fa:sitemap"></iconify-icon> Hardin, J., Rocke, D.;&nbsp;<strong><a href="http://pages.pomona.edu/~jsh04747/Research/ROBDIST_final.pdf" target="_blank">The&nbsp;Distribution of Robust Distances</a></strong>,&nbsp;<em>Journal of Computational and Graphical Statistics</em>,&nbsp;<em>14</em>: 1-19, 2005.&nbsp;&nbsp;(R code: to estimate the MCD ‚Äì&nbsp;<a href="http://pages.pomona.edu/~jsh04747/Research/mcd.est.r" target="_blank">mcd.est.r</a>&nbsp;and to estimate c and m ‚Äì&nbsp;<a href="http://pages.pomona.edu/~jsh04747/Research/cm.r" target="_blank">cm.r</a>.)
</li>
<li>
<iconify-icon inline="" icon="fa-solid:chart-line"></iconify-icon> Hardin, J.;&nbsp;<strong><a href="http://pages.pomona.edu/~jsh04747/Research/Hardin%20Article%20(WINTER%202005).pdf" target="_blank">Microarray Data from a Statistician‚Äôs Point of View</a></strong>,&nbsp;<em>STATS</em>,<em>&nbsp;42</em>:4-13, 2005.
</li>
</ul>
<p>&nbsp;</p>
<strong>2004</strong>
<ul>
<li>
<iconify-icon inline="" icon="fa-solid:chart-line"></iconify-icon> Hardin, J., Waddell, M., Page, D., Zhan, F.,&nbsp;Barlogie, B., Crowley, J.,&nbsp;Shaughnessy, J.;&nbsp;<strong><a href="http://www.bepress.com/sagmb/vol3/iss1/art10/" target="_blank">Evaluation of Multiple Models to Distinguish Closely Related Forms of Disease Using DNA Microarray Data: an Application to Multiple Myeloma</a></strong>,&nbsp;&nbsp;<em>Statistical Applications in Genetics and Molecular Biology,</em>&nbsp;<em>3 (</em>article 10), 2004.
</li>
<li>
<iconify-icon inline="" icon="fa:sitemap"></iconify-icon> Hardin, J., Rocke, D.;&nbsp;<strong>&nbsp;</strong><strong><a href="http://pages.pomona.edu/~jsh04747/Research/clstout3.PDF" target="_blank">Outlier Detection in the Multiple Cluster Setting Using the Minimum Covariance Determinant Estimator</a></strong>,&nbsp;<em>Computational Statistics and Data Analysis</em>,&nbsp;<em>44</em>: 625-638, 2004.
</li>
<li>
<iconify-icon inline="" icon="fa-solid:chart-line"></iconify-icon> Pauler, D., Hardin, J., Faulkner, J., Leblanc, M., Crowley, J.;&nbsp;<strong>Survival Analysis with Gene Expression Arrays</strong>.&nbsp;In&nbsp;<em><a href="http://www.elsevier.com/wps/find/bookdescription.cws_home/699537/description#description" target="_blank">Handbook of Statistics 23: Advances in Survival Analysis,</a></em>&nbsp;eds.&nbsp;N.Balakrishnan&nbsp;and C.R.&nbsp;Rao, Elsevier Science:&nbsp;Amsterdam; 2004.
</li>
</ul>
<p>&nbsp;</p>
<strong>2002</strong>
<ul>
<li>
<iconify-icon inline="" icon="fa-solid:chart-line"></iconify-icon> Durbin, B., Hardin, J., Hawkins, D., Rocke, D.;&nbsp;<strong>&nbsp;</strong><strong><a href="http://bioinformatics.oxfordjournals.org/content/18/suppl_1/S105" target="_blank">A Variance-Stabilizing Transformation for Gene-Expression Microarray Data</a></strong>;&nbsp;<em>Bioinformatics</em>,&nbsp;<em>18</em>: S105-S110, 2002.
</li>
<li>
<iconify-icon inline="" icon="fa-solid:chart-line"></iconify-icon> Zhan, F., Hardin, J.,&nbsp;Kordsmeier, B.,&nbsp;Bumm, K.,&nbsp;Zheng, M., Tian, E., Sanderson, R., Yang, Y., Wilson, C.,&nbsp;Zangari, M.,&nbsp;Anaissie, E., Morris, C.,&nbsp;Muwalla, F., van Rhee, F.,&nbsp;Fassas, A., Crowley, J., Tricot, G.,&nbsp;Barlogie, B.,&nbsp;Shaughnessy, J.;&nbsp;<strong><a href="http://www.bloodjournal.org/cgi/reprint/99/5/1745.pdf" target="_blank">Global Gene Expression Profiling of Multiple Myeloma, Monoclonal&nbsp;Gammopathy&nbsp;of Undetermined Significance, and Normal Bone Marrow Plasma Cells</a></strong>;&nbsp;<em>Blood</em>,&nbsp;<em>99</em>: 1745-1757, 2002.
</li>
</ul>
<p>&nbsp;</p>
<strong>1999-2000</strong>
<ul>
<li>
<iconify-icon inline="" icon="fa:sitemap"></iconify-icon> Hardin, J.;&nbsp;<strong><a href="http://pages.pomona.edu/~jsh04747/Research/t.pdf" target="_blank">Multivariate Outlier Detection and Robust Clustering with Minimum Covariance Determinant Estimation and S-Estimation</a></strong>.&nbsp;&nbsp;Ph.D.&nbsp;thesis, Statistics; University of California, Davis.&nbsp;&nbsp;2000.
</li>
<li>
<iconify-icon inline="" icon="fa:sitemap"></iconify-icon> Coleman, D., Dong, X., Hardin, J., Rocke, D.M., Woodruff, D.L.;&nbsp;<a href="http://research.pomona.edu/johardin/files/2012/08/Comp6.pdf" target="_blank"><strong>Some Computational Issues in Cluster Analysis with no √† priori Metric</strong></a>;&nbsp;<em>Computational Statistics and Data Analysis</em>,&nbsp;<em>31</em>: 1-11, 1999.
</li>
</ul>
<p>&nbsp;</p>



</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "Óßã";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp("https:\/\/hardin47\.netlify\.app\/");
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->
<footer class="footer">
  <div class="nav-footer">
    <div class="nav-footer-left">
<p>¬© Copyright 2025, Jo Hardin</p>
</div>   
    <div class="nav-footer-center">
      &nbsp;
    </div>
    <div class="nav-footer-right">
<p>Built with üíö and <a href="https://quarto.org/">Quarto</a>.</p>
</div>
  </div>
</footer>




</body></html>